### ISSUE TITLE
[a11y]: Overflow Menu - VO - User unable to close menu

### ISSUE THREAD

**Issue by annawen1** [carbon-design-system/carbon#11083]
### Package

carbon-components-react

### Browser

Safari

### Operating System

MacOS

### Package version

7.56.0

### React version

16.9.0

### Automated testing tool and ruleset

Manual testing

### Assistive technology

iPhone 12 - VoiceOver

### Description

When user opens the Overflow Menu and swipes through menu items, focus goes out of menu but also when user swipes back up through the menu items, the focus trapping prevents user from reaching the trigger element to close the menu.

Expecting: Focus should be trapped within the menu and allow for user to go back to the trigger element to close menu.

https://user-images.githubusercontent.com/54281166/160434667-dcb6c357-fb97-40b3-ac09-e101b7d62b74.mov

### CodeSandbox example

N/A

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default storybook in mobile device with VO
2. double tap to open overflow menu
3. swipe through menu items, notice that focus goes outside of menu
4. swipe back up through menu items and try to close the menu 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Potentially related: https://github.com/carbon-design-system/carbon/issues/13115

**Comment by tw15egan** [comment]
The main issue (inability to close the menu) has been resolved. I don't believe the Voiceover functionality to move between items on the device should be trapped like it is with focus, as it is searching for the next element on the entire device that should be read next.

https://github.com/carbon-design-system/carbon/assets/11928039/cc19ae6e-0fba-4696-9892-a5c0e0bfef8e
